verification/verify.go

```diff
@@ -153,3 +163,80 @@ func IsRetryable(err error) bool {
 	_, retryable := err.(Retryable)
 	return retryable
 }
+
+func verifyPixelParams(params *Params) error {
+	// check if the node is on-chain mode
+	if params != nil &&
+		params.Orchestrator != nil &&
+		params.Orchestrator.TicketParams != nil {
+
+		var err error
+		for i := 0; err == nil && i < len(params.Results.Segments); i++ {
+			if err := verifyPixels(
+				params.Results.Segments[i].Url,
+				flatten(params.Renditions),
+				params.Results.Segments[i].Pixels); err != nil {
+
+				glog.Error(err)
+				return ErrPixelMismatch
+			}
+		}
+		return nil
+
+	}
+	return ErrPixelsAbsent
+}
+
+func flatten(s [][]byte) (r []byte) {
+	for _, e := range s {
+		for _, f := range e {
+			r = append(r, f)
+		}
+	}
+	return
+}
+
+func verifyPixels(fname string, data []byte, reportedPixels int64) error {
+	uri, err := url.ParseRequestURI(fname)
+	// If the filename is a relative URI and the broadcaster is using local memory storage
+	// write the data to a temp file
+	if err == nil && !uri.IsAbs() && data != nil {
+		tempfile, err := ioutil.TempFile("", common.RandName())
+		if err != nil {
+			return fmt.Errorf("error creating temp file for pixels verification: %v", err)
+		}
+		defer os.Remove(tempfile.Name())
+
+		if _, err := tempfile.Write(data); err != nil {
+			tempfile.Close()
+			return fmt.Errorf("error writing temp file for pixels verification: %v", err)
+		}
+
+		if err = tempfile.Close(); err != nil {
```

Comment:
Could we call `Close()` with a `defer` after the `defer` statement that calls `os.Remove()` to avoid calling it in 2 places (this is what was done in this function before this PR)?

Reply:
That was a bug and I fixed it.  It is incorrect, bad practice to defer close() on writeable files
https://www.joeshaw.org/dont-defer-close-on-writable-files

Reply:
I think the previous usage was probably ok given the current implementation of `file.Close()` and the surrounding code, but agreed that always handling errors even in such a situation is better. 

We could use defer here *and* handle the error within the defer, but I can understand a dislike for that pattern, so I'm ok with how the `Close()` calls are currently structured.